Dockerfile

```diff
@@ -4,66 +4,94 @@ ARG RUBY_VERSION=3.4.4
 
 FROM docker.io/library/ruby:$RUBY_VERSION-slim AS base
 
-ENV RACK_ENV=production
-ENV HANAMI_ENV=production
-ENV HANAMI_SERVE_ASSETS=true
-ENV BUNDLE_DEPLOYMENT=1
 ENV BUNDLE_PATH=/usr/local/bundle
-ENV BUNDLE_WITHOUT="development:quality:test:tools"
-
 WORKDIR /app
 
+# Combined repository setup and package installation
 RUN <<STEPS
-  apt-get update -qq \
-  && apt-get install --no-install-recommends -y \
-  chromium \
-  curl \
-  imagemagick \
-  libjemalloc2 \
-  nodejs \
-  npm \
-  postgresql-client \
-  tmux \
-  && rm -rf /var/lib/apt/lists /var/cache/apt/archives
+  apt-get update -qq
+  apt-get install -y --no-install-recommends curl ca-certificates gnupg2 lsb-release
+  # Setup PostgreSQL repository
+  curl -fsSL https://www.postgresql.org/media/keys/ACCC4CF8.asc | gpg --dearmor -o /usr/share/keyrings/postgresql-keyring.gpg
+  echo "deb [signed-by=/usr/share/keyrings/postgresql-keyring.gpg] https://apt.postgresql.org/pub/repos/apt $(lsb_release -cs)-pgdg main" > /etc/apt/sources.list.d/pgdg.list
+  # Setup Node.js repository and install runtime dependencies
+  curl -fsSL https://deb.nodesource.com/setup_23.x | bash -
```

Comment:
:thought_balloon: Hmm, is this the official way to install the most recent version of Node.js? In fact, this app is using Node 24.1.0 so if we go this route, we'd probably need a dynamic way to pick up the proper Node install based on the `.node-version`?

Reply:
this image only has the repository for node 18 so yes this would be how we add the repository for the version of node this app requires. the image out of the box has node 18 installed

![Screenshot 2025-05-27 at 11 03 57 AM](https://github.com/user-attachments/assets/45afe01e-524c-4425-8348-1f9f07fb2e7a)

Totally agree it should be dynamic to the version of node the app requires

I don't know if the app needs nodejs at runtime (im guessing liklely not) as the assets are built at build time so this was just to have the proper software for development on the container